src/Microsoft.AspNetCore.Blazor.Razor.Extensions/BlazorCodeTarget.cs

```diff
@@ -11,11 +13,47 @@ namespace Microsoft.AspNetCore.Blazor.Razor
     /// </summary>
     internal class BlazorCodeTarget : CodeTarget
     {
+        private readonly RazorCodeGenerationOptions _options;
+
+        public BlazorCodeTarget(RazorCodeGenerationOptions options, IEnumerable<ICodeTargetExtension> extensions)
+        {
+            _options = options;
+            Extensions = extensions.ToArray();
+        }
+
+        public ICodeTargetExtension[] Extensions { get; }
+
         public override IntermediateNodeWriter CreateNodeWriter()
-            => new BlazorIntermediateNodeWriter();
+        {
+            return _options.DesignTime ? (IntermediateNodeWriter)new DesignTimeNodeWriter() : new BlazorIntermediateNodeWriter();
+        }
```

Comment:
We need to properly implement codetarget for tooling. Looking for a way to make this not needed, but it seems OK for now.